insolar/record/record.go

```diff
@@ -0,0 +1,246 @@
+//
+// Copyright 2019 Insolar Technologies GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//go:generate protoc -I./vendor -I./ --gogoslick_out=./ insolar/record/record.proto
+package record
+
+import (
+	"github.com/insolar/insolar/insolar"
+	"io"
+	"reflect"
+
+	"github.com/pkg/errors"
+)
+
+type SerializableRecord interface {
+	MarshalRecord() ([]byte, error)
+	MarshalRecordTo(data []byte) (int, error)
```

Comment:
Do we really need this method?

Reply:
It allows us to encode message into existing buffer, and will give us ability to "control" maximum message output size.